packages/node_modules/@ciscospark/internal-plugin-user/src/user.js

```diff
@@ -289,10 +322,57 @@ const User = SparkPlugin.extend({
         if (res.body.hasPassword || res.body.sso) {
           this.hasPassword = true;
         }
+
+        // if we have a different CI than default, and Federation is enabled,
+        // update the CI to what was returned by Atlas.
+        if (this.spark.config.credentials.federation) {
+          this.spark.config.credentials.idbroker.url = res.body.userEntities.idBrokerUrl;
+          this.spark.config.credentials.identity.url = res.body.userEntities.identityUrl;
+        }
+
         return res.body;
+      })
+      .catch((reason) => {
+        // reset CI on failure
+        if (this.spark.config.credentials.federation) {
+          this.spark.config.credentials.idbroker.url = orgIdbrokerUrl;
+          this.spark.config.credentials.identity.url = orgIdentityUrl;
+        }
+        return Promise.reject(reason);
       });
   },
 
+
+  /**
+   * If the passed-in condition is true, retrieve the user's
+   * CI and update config.
+   * Otherwise, do nothing, leave currently configured CI.
+   * @param {string} email (required)
+   * @param {boolean} condition (required)
```

Comment:
Let's rename `condition` to what it does. Maybe `suppressEmail`?

Reply:
It's whether we lookup the user's CI or just return what is already config'd.  Changed it to reflect that it triggers the CI lookup.